### DISCUSSION TITLE
firstOrNew in WebHookHandler

### DISCUSSION THREAD

**Discussion by ah-rahimi** [def-studio/telegraph#141] [General]
the firstOrNew in WebHookHandler not working
![image](https://user-images.githubusercontent.com/87445571/184992654-3d89d51f-d155-4887-a503-62f566b7bf52.png)
you must use firstOrCreate or better: UpdateOrCreate
& how we can add field to it?

**Comment by fabio-ivona** [comment]
What field do you want to add?

**Reply by ah-rahimi** [reply]
and now don't add to database

**Comment by ah-rahimi** [comment]
first_name
last_name
lang
user_name
and... for my bot

**Comment by fabio-ivona** [comment]
You should implement your own bot model and set it in config file

**Reply by ah-rahimi** [reply]
it's true
But how to tell him to check if there is a user or to update when he receives the request
And how to give new fields to it?

**Reply by ah-rahimi** [reply]
@fabio-ivona 
i create PR for this
that i can customize it
please see: #142

**Reply by fabio-ivona** [reply]
I'm planning to add events for such handling, no need to override that method

On 22nd of August I'll be back from vacation and will take care of this

**Reply by ah-rahimi** [reply]
Excellent, thank you

**Reply by ah-rahimi** [reply]
But for now, I solved the problem in my own way :)

![image](https://user-images.githubusercontent.com/87445571/185362258-f057833d-684b-459c-a1f0-3876821babc1.png)

**Reply by fabio-ivona** [reply]
This is a clean and good solution too

**Comment by ah-rahimi** [comment]
It is a very good package
It has only a few general problems that everything is in one section
The name and username of the user should be taken from one section
Chat ID should be obtained from a section
Each is the same, and the documentation is a bit confusing
However, it has good features

**Reply by fabio-ivona** [reply]
This maps telegram data structure

**Comment by ah-rahimi** [comment]
one question: @fabio-ivona 
How to recognize and take the value of the front of the start?
@username?start=somethings

**Reply by fabio-ivona** [reply]
Sorry can't understand what you are asking

**Reply by ah-rahimi** [reply]
if user click on 
t.me/username?start=some-text
send request text like:  /start some-text
how get some-text??

**Reply by ah-rahimi** [reply]
@fabio-ivona ?

**Reply by fabio-ivona** [reply]
Command parameters will be supported soon

**Reply by fabio-ivona** [reply]
@ahr1384 command parameters are supported in the upcoming version, you can retrieve the text after the command as a parameter of the handler function:

```
\hi Fabio
```

```php
class CustomWebhookHandler extends WebhookHandler
{
    public function hi(string $userName)
    {
        $this->chat->markdown("*Hi* $userName, happy to be here!")->send(); // Hi Fabio, happy to be here!
    }
}
```

**Comment by fabio-ivona** [comment]
as this was a breaking change, this behaviour has been reverted in #161

see the new config file to enable messages from unknown chats
